Assignment5/FindAlphabet.java

```diff
@@ -0,0 +1,71 @@
+package Assignment5;
+
+import java.util.ArrayList;
+
+public class FindAlphabet {
+
+	public static String findAlphabet(ArrayList<String> dictionary) {
+		/* The dictionary contains only one word */
+		if (dictionary.size() == 1) {
+			return dictionary.get(0);
+		}
+
+		Graph graph = new Graph();
+
+		String previousWord = dictionary.get(0);
+
+		 /* Compare each word from the dictionary with the previous one in order to add
+		  * edges in the graph (which will represent alphabet's constraints) */
+		for (String currentWord: dictionary) {
+
+			/* Invalid input (if the current word is a prefix of the previous word, then the dictionary is invalid */
+			if (previousWord.length() > currentWord.length() && previousWord.startsWith(currentWord) == true)
+				return null;

```

Comment:
Throwing an Exception would be much better java style here, rather than returning null. Maybe an IllegalArgumentException or a custom exception.